### DISCUSSION TITLE
Scoping the options, Relation Managers with dynamic values

### DISCUSSION THREAD

**Discussion by KONEY** [filamentphp/filament#5995] [Help]
hi,

I'm trying to prevent to attach multiple times the same record by filtering it out from the available options.

Docs https://filamentphp.com/docs/2.x/admin/resources/relation-managers#scoping-the-options suggest to pass a query like **AttachAction::make()
    ->recordSelectOptionsQuery(fn (Builder $query) => $query->whereBelongsTo(auth()->user())** but it's not clear how to pass the current record in order to return only records not already attached.

Any help or hint?
Thanks!

**Comment by danharrin** [comment]
It should already prevent duplicates though, we will search which records have already been attached

**Reply by danharrin** [reply]
Unless you have `allowDuplicates` enabled

**Reply by KONEY** [reply]
Thanks! But this is not happening, having allowDuplicates not enabled.

If it's supposed to be like this I'll investigate further. Anyway if you have any idea why this isn't triggered you're welcome to share.

Regards

**Reply by danharrin** [reply]
I would advise debugging the code we use internally to work out why it doesn't work for you

**Comment by KONEY** [comment]
It looks like the subquery is failing for some reason and always returning all records. Going back to the initial post, is the actual values available somehow inside this scope?

Thanks

**Reply by danharrin** [reply]
Its best if we work out how to fix the issue instead of plastering over it. Please open an issue with a reproduction repository for us to use, or a PR to fix.

**Reply by KONEY** [reply]
Sure, thanks. That was not my intention, someone else is on it and I'm trying to figure out a backup solution.

**Reply by KONEY** [reply]
I created an identical relationship which works as is supposed to. Turns out slightly different subqueries are built:

first relationship:

> select * from `catalogue_items` where `is_accessory` = 1 and not exists (select * from `catalogue_items` as `laravel_reserved_0` inner join `compatibilities_pivot` on `laravel_reserved_0`.`id` = `compatibilities_pivot`.`id_item` where `catalogue_items`.`id` = `compatibilities_pivot`.`id_assoc` and `catalogue_items`.`id` = 40 and `is_accessory` = 0 and `laravel_reserved_0`.`deleted_at` is null) and `catalogue_items`.`deleted_at` is null order by `code` asc

second, working relationship:

> `select * from `users` where not exists (select * from `catalogue_items` inner join `dummy_pivot` on `catalogue_items`.`id` = `dummy_pivot`.`id_item` where `users`.`id` = `dummy_pivot`.`id_assoc` and `catalogue_items`.`id` = 40 and `is_accessory` = 0 and `catalogue_items`.`deleted_at` is null) and `users`.`deleted_at` is null order by `first_name` asc`

So it looks like an alias `laravel_reserved_0` for the join is added and I belive is what makes the query not filtering results as expected. But I'm having a hard time figuring out why they end up with such a difference.

**Reply by KONEY** [reply]
Figured out what the problem is: the two models in relationship share the same table and this triggers a bug (maybe in eloquent, don't know yet) in which an alias (laravel_reserved_0) is added to the subquery's join which makes it fail and always return all records.

**Reply by danharrin** [reply]
potentially sounds like an Eloquent bug, yes

**Reply by KONEY** [reply]
Indeed. I may really use a backup way of filtering the results inside the scope :)

**Comment by aurawindsurfing** [comment]


**Reply by KONEY** [reply]
Thanks. Actually it completely mismatches with my relationship structure so I'm having some troubles trying to figure out how to implement and test it.
